soroban-sdk/src/auth.rs

```diff
@@ -1,15 +1,64 @@
 use crate::{contracttype, Address, BytesN, RawVal, Symbol, Vec};
 
+/// Context of a single authorized call peformed by an address.
+///
+/// Custom account contracts that implement `__check_auth` special function
+/// receive a list of `Context` values corresponding to all the calls that
+/// need to be authorized.
+#[derive(Clone)]
 #[contracttype(crate_path = "crate", export = false)]
+pub enum Context {
```

Comment:
I thought I saw a comment about it being a bit strange we are using contracttype for built-in stuff. I actually think this is good. It forces us to ensure the contracttype is efficient, and reduces the number of bespoke host functions we need.

Reply:
Yeah, I can see both pros and cons to this, but ultimately it makes things simpler and allows for adding more complex host fns or more complex built-in contracts.